ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSbasic.cpp

```diff
@@ -24,138 +24,134 @@ using std::string;
 #include "Audio_Systems/audio_mandatory.h"
 #include "DSsystem.h"
 
-#ifdef DEBUG_MODE
-#include "Widget_Systems/widgets_mandatory.h"  // show_error
-#include "libEGMstd.h"
-#endif
-
 #include "Universal_System/estring.h"
 
 namespace enigma_user {
 
-bool sound_exists(int sound) { return unsigned(sound) < sound_resources.size() && bool(sound_resources[sound]); }
+bool sound_exists(int sound) { return sound_resources.exists(sound); }
 
 bool sound_play(int sound)  // Returns whether sound is playing
 {
-  get_sound(snd, sound, false);
-  snd->soundBuffer->SetCurrentPosition(0);
-  snd->soundBuffer->Play(0, 0, 0);
+  auto snd = sound_resources.get(sound);
```

Comment:
May be better to use the real type instead of `auto`, but I guess this doesn't hurt.

Reply:
Why?

Reply:
This is interesting, you know I thought of something we could do. We could actually create inlined member setters and getters for the asset array. That's typically what will be done with the assets in the array more than half of the time. If we just had a member setter and getter, we wouldn't really need to declare a local the majority of the time.

Reply:
You don't have to use the full type, but at least put the * or & required to make it clear that no copy is being performed. Sometimes using auto can make code harder to edit because the actual type of an object isn't apparent (or even just the spelling).

Reply:
Fair enough: 79faa632cd4bc497a38a4d823be4bce3b830e0a1 & 161057c4bcc1beb1cbf698c7df0cf9f10c725729

No constant correctness is supported yet since we don't see a need for it.